web_drop_target/static/src/js/web_drop_target.js

```diff
@@ -0,0 +1,132 @@
+            var result = this._super.apply(this, arguments);
+            this.$el.on('drop.widget_events', this.proxy('_on_drop'));
+            this.$el.on('dragenter.widget_events', this.proxy('_on_dragenter'));
+            this.$el.on('dragover.widget_events', this.proxy('_on_dragenter'));
+            this.$el.on('dragleave.widget_events', this.proxy('_on_dragleave'));
+            return result;
+        },
+
+        _on_drop: function(e) {
+            var drop_item = this._get_drop_item(e);
+            if(!drop_item) {
+                return;
+            }
+            jQuery(e.delegateTarget).removeClass(this._drag_over_class);
+            var reader = new FileReader();
+            reader.onloadend = this.proxy(
+                _.partial(this._handle_file_drop, drop_item.getAsFile())
+            );
+            reader.readAsArrayBuffer(drop_item.getAsFile());
+            e.preventDefault();
+        },
+
+        _on_dragenter: function(e) {
+            if(this._get_drop_item(e)) {
+                e.preventDefault();
+                jQuery(e.delegateTarget).addClass(this._drag_over_class);
+                return false;
+            };
+        },
+
+        _on_dragleave: function(e) {
+            jQuery(e.delegateTarget).removeClass(this._drag_over_class);
+        },
+
+        _get_drop_item: function(e) {
+            var self = this,
+                dataTransfer = e.originalEvent.dataTransfer,
+                drop_item = null;
+            _.each(dataTransfer.items, function(item) {
+                if(
+                    _.contains(self._drop_allowed_types, item.type) ||
+                    _.isEmpty(self._drop_allowed_types)
+                ) {
+                    drop_item = item;
+                }
+            });
+            return drop_item;
+        },
+
+        _handle_file_drop: function(drop_file, e) {
+            // do something here, for example call the helper function below
+            // e is the on_load_end handler for the FileReader above,
+            // so e.target.result contains an ArrayBuffer of the data
+        },
+
+        _handle_file_drop_attach: function(
+                drop_file, e, res_model, res_id, extra_data
+        ) {
+            // helper to upload an attachment and update the sidebar
+            var self = this;
+            return new Model('ir.attachment').call(
+                'create',
+                [
+                    _.extend({
+                        name: drop_file.name,
+                        datas: base64js.fromByteArray(
+                            new Uint8Array(e.target.result)
+                        ),
+                        datas_fname: drop_file.name,
+                        res_model: res_model,
+                        res_id: res_id,
+                    }, extra_data || {})
+                ]
+            )
+            .then(function() {
+                // try to find a sidebar and update it if we found one
```

Comment:
Can we add a notification once dropped file is attached and attachment created successfully ?

Reply:
makes a lot of sense. But as this is an extra feature, I'd prefer to wrap this up and have whomever add this who needs it

Reply:
:+1:

Reply:
you could fire a proper event (via `trigger_up`?), then anybody can hook to it if needed

Reply:
can you drop a comment at least or few lines into ROADMAP to not forget these improvements? :wink